src/main/scala/dev/kamu/cli/commands/PullCommand.scala

```diff
@@ -20,15 +20,44 @@ class PullCommand(
   private val logger = LogManager.getLogger(getClass.getName)
 
   def run(): Unit = {
-    val datasets =
-      if (all)
-        metadataRepository.getAllDatasets()
-      else
-        datasetIDs
-          .map(DatasetID)
-          .map(metadataRepository.getDataset)
-
-    val numUpdated = datasets
+    val s = metadataRepository.getAllDatasetIDs()
+    val datasets: IndexedSeq[DatasetID] = s.toIndexedSeq
+    val datasetIntID = s.zipWithIndex.toMap
+    val idToInt: DatasetID => Int = d => datasetIntID(d)
```

Comment:
can be written more concisely as:
`val idToInt = (id: DatasetID) => datasetIntID(d)`
or even
`val idToInt = datasetIntID.apply _`